src/simulations/collect.jl

```diff
@@ -174,38 +181,60 @@ function init_agent_dataframe(model::ABM, properties::Vector{<:Tuple})
     headers[1] = "step"
     types[1] = Int[]
 
-    for (i, (k, agg)) in enumerate(properties)
-        headers[i+1] = aggname(k, agg)
+    for (i, property) in enumerate(properties)
+        k, agg = property
+        headers[i+1] = aggname(property)
         # This line assumes that `agg` can work with iterators directly
         current_type = typeof( agg( get_data(a, k) for a in Iterators.take(alla,1) ) )
         isconcretetype(current_type) || warn("Type is not concrete when using function $(agg) "*
-        "on key $(k). Consider using type annotation, e.g. $(agg)(a)::Float64 = ...")
+            "on key $(k). Consider using type annotation, e.g. $(agg)(a)::Float64 = ...")
         types[i+1] = current_type[]
     end
     DataFrame(types, headers)
 end
 
 """
-    aggname(k) → name
+    aggname(k; condition = false) → name
 
 Return the name of the column of the `i`-th collected data where `k = adata[i]`
 (or `mdata[i]`).
+
+Passing tuple types is the suggested method such that `(:weight, mean)` and
+`(:weight, mean, cond)` can return appropreate columns.
+To return a conditional aggregate with separate variables, use
+`aggname(k, agg; condition = true)`.
 """
-aggname(k, agg) = string(agg)*"_"*string(k)
-aggname(x::Tuple) = aggname(x[1], x[2])
+function aggname(k, agg; condition = false)
+    if condition
+        "conditional_"*string(agg)*"_"*string(k)
+    else
+        string(agg)*"_"*string(k)
+    end
+end
+aggname(x::Tuple{K,A}) where {K,A} = aggname(x[1], x[2])
+aggname(x::Tuple{K,A,<:Function}) where {K,A} = aggname(x[1], x[2]; condition = true)
 aggname(x::Union{Function, Symbol, String}) = string(x)
 
 function collect_agent_data!(df, model, properties::Vector{<:Tuple}, step::Int=0)
     alla = allagents(model)
     push!(df[!, 1], step)
-    for (i, (k, agg)) in enumerate(properties)
-        _add_col_data!(df[!, i+1], agg, k, alla)
+    for (i, prop) in enumerate(properties)
+        _add_col_data!(df[!, i+1], prop, alla)
     end
     return df
 end
-# Function barrier
-function _add_col_data!(col::AbstractVector{T}, agg, k, alla) where {T}
-    res::T = agg(get_data(a, k) for a in alla)
+
+# Normal aggregates
+function _add_col_data!(col::AbstractVector{T}, property::Tuple{K,A}, agent_iter) where {T,K,A}
+    k, agg = property
+    res::T = agg(get_data(a, k) for a in agent_iter)
+    push!(col, res)
+end
+
+# Conditional aggregates
+function _add_col_data!(col::AbstractVector{T}, property::Tuple{K,A,<:Function}, agent_iter) where {T,K,A}
```

Comment:
```suggestion
function _add_col_data!(col::AbstractVector{T}, property::Tuple{K,A,F}, agent_iter) where {T,K,A,F}
```

Limiting to `Function` excludes functors from being used (very unlikely, but also there is no gain in putting `<:Function` instead of `<:Any`)